scripts/util.py

```diff
@@ -153,6 +154,67 @@
 ] = lambda self, data: yaml.representer.SafeRepresenter.represent_str(self, str(data))
 
 
+class ApiEndpoint(Enum):
+    COMPUTE = "compute"
+    BQ = "bq"
+    STORAGE = "storage"
+    TPU = "tpu"
+    SECRET = "secret_manager"
+
+
+def default_credentials():
+    return google.auth.default()[0]
+
+
+def authentication_project():
+    return google.auth.default()[1]
+
+
+def universe_domain() -> str:
+    return instance_metadata("attributes/universe_domain")
+
+
+def endpoint_version(api: ApiEndpoint) -> Optional[str]:
```

Comment:
Either make it `Optional` or don't handle possibility of `None`.

It it's `Optional`, than update `create_client_options` by removing needles `if api:`